### PR TITLE
Introduce partial update for checkout

### PR DESCRIPTION
**Pull request by sav007**
@krisorr

### CODE REVIEW

File: MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutService.java [hunk 1]
```diff
@@ -61,47 +61,19 @@
...
-     */
-    Observable<Checkout> updateCheckoutShippingRate(String checkoutToken, ShippingRate shippingRate);
-
-    /**
-     * Update an existing Checkout's attributes
-     *
-     * @param checkout the {@link Checkout} to update
+     * Update an existing Checkout's attributes.
+     *
+     * Only attributes from the list below are going to be updated, any other will be ignored:
```
Comment (krisorr) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutService.java]: Only the following attributes will be updates, any others will be ignored:

File: MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutService.java [hunk 2]
```diff
@@ -61,47 +61,19 @@
...
+     * <ul>
+     * <li>email</li>
+     * <li>shipping address</li>
```
Comment (krisorr) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutService.java]: should reference the actual member variable in the javadoc

File: MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java [hunk 3]
```diff
@@ -180,9 +146,20 @@ public CancellableTask updateCheckout(final Checkout checkout, final Callback<Ch
             throw new NullPointerException("checkout cannot be null");
         }
 
-        final Checkout safeCheckout = checkout.copyForUpdate();
```
Comment (krisorr) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java]: can we get rid of copyForUpdate() now?

Comment (sav007) [inline MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/CheckoutServiceDefault.java]: Yep

**Comment by krisorr** [comment]
LGTM if you all the tests except for Customer pass (there is an known issue with Customer test right now)

**Pull request merged**